### ISSUE TITLE
Sprint May 29 to June 9

### ISSUE THREAD

**Issue by mariusandra** [PostHog/posthog#15690]
# Global Sprint Planning

## 3 things that might take us down

1.
2.
3.

## Retro: What can we do better next sprint?

1.
2.
3.
4.
5.

# Team sprint planning

For your team sprint planning copy this template into a comment below for each team.

```
# Team ___

**Support hero:** ___

## Retro

<!-- Grab the high and low priority items from last time and add whether that item was completed or not -->

- 

## Hang over items from previous sprint

<!-- For each item, decide to re-prioritise (and add below) or deprioritise -->

- Item 1. prioritised/deprioritise

## OKR

1. OKR, status (red/yellow/green) and action points if yellow/red


### High priority

-

### Low priority / side quests

-

```

**Comment by pauldambra** [comment]
# Next sprint - Team Moni-(a bird in the hand is worth)-to-(in the bush)-ring

**Support hero:** @pauldambra , then @benjackwhite 

## Retro

- 

## Hang over items from previous sprint

* Notebooks MVP - waiting for DB deployment
* S3 Replay stuff - last confounding bugs

## Goals
* Goal 1 : Finish the 80:20 of session recording 🟢
  * S3 work is on track it feels
* Goal 2 : Notebooks MVP 🟢
  * Starting to shape up - Insights are the most painful part
* Goal 3 : Release MVP of Error tracking product 🟡
  * Not started - delayed another sprint to make sure we don't end up too split

### High priority

- Autocapture for frontend Exceptions & Improved Replay filtering (Exceptions, active time and more...) @benjackwhite  and @pauldambra 
- Notebooks filtering widget + feedback on Notebooks @benjackwhite @pauldambra 

### Low priority / side quests

- Turn replay listing on for everyone @pauldambra 
- S3 playback in early access @pauldambra 
- Agree on Automation MVP @benjackwhite

**Comment by mariusandra** [comment]
# Team Product Analytics

Captain Toads next 2 weeks:
- @yakkomajuri first week
- @Twixes second week

## Retro

- @thmsobrmlr support consistently takes up a lot of time
- @Twixes it's a good thing that there are people who want to report things. Spent all of last week fixing thing (support leftover)
- @yakkomajuri HouseWatch was good
- @mariusandra sick/off/support, limited work, blocked with releasing HogQL

## Hang over items from previous sprint

- PoEv2
- PostHog 3000
- Pentest

## OKR

- 🌞 Objective 1: PostHog 3000 Hacker News launch is a wild success
- 🌞 Objective 2: Make PostHog performance frustration free for our 10 largest customers
- 🍏 Objective 3: HogQL


### High priority

- Ongoing support
- HogQL release and any followup @mariusandra 
- PostHog 3000 insights UI @Twixes (support 2nd week)
- PoEv2 @yakkomajuri (support 1st week)

### Low priority / side quests

- Removing all legacy insight components & data exploration followup @thmsobrmlr

**Comment by raquelmsmith** [comment]
# "Team" Growth

## Retro


- [x] Test out new product + group analytics pricing
- [x] Improvements for CS
  - Free trials - is usage still tracked when they are on a free trial? What happens when they go over the usage limits for the plan they are on?
  - Cancelling (or not) subs for unpaid invoices https://github.com/PostHog/company-internal/issues/1019
- [x] Move all existing customers to new billing-by-products UI
  - I think I can do this today.
- [x] Talk to customers about free plans - calls scheduled this and next week
- [ ] Create modal for prompting to upgrade to addons when sub-ing to a main product
  - Deprioritized this for now but added it to the Low Priority section
- Generally got sidelined this sprint with a number of billing bugs. All fixed now. 

## OKR

1. We can easily run experiments on pricing 🟡 
    - Not sure I can do this and the other one just by myself, but I will do my best. We decided to prioritize the other one first.
2. Each paid product is positioned as a first-class citizen within our onboarding and upgrade flows 🟡 
    - I think I can make good progress toward this by the end of the quarter. We’re now in a good spot to do this since the switch to separate upgrades per product is done.
    - Planning issue: https://github.com/PostHog/posthog/issues/15651

## This sprint
### High priority

- [x] Make a template for empty states on any product and apply to the products.
- [ ] Create a template/scene that each product can use to introduce people to the product before they start using it (eg a landing/marketing page, but in-app).

### Low priority / side quests

- [ ] Create modal for prompting to upgrade to addons when sub-ing to a main product
- [ ] Improve modal for upgrading to paid features
  - Let them subscribe in the modal
  - Tell them which product they need

**Comment by tiina303** [comment]
# 🚰 Team Pipeline 🚰 

Support heros next 2 weeks:  Harry, then Tiina
Vacation: Tomas

## Retro
- @hazzadous - incidents have been annoying, feel much better about my understanding of temporal, good to get out stuff
- @tiina303 - happy about killing lazy person container and redis person caching. Maybe redis will be happier now. Opens way to resolve processing issues. Sorry for me to have been on top of incidents and thanks Harry.
- @james - lots of outages, we're going round-robin on different issues causing it. Stuff that we're doing is making things more resilient. We know what the next steps are feels good.
- @xvello - happy about batching, happy about pending rdkafka, London trip + Ellie 👌😎 2 days 3 incidents still distracting

What will bring us down

Redis doesn’t look happy. PostHog doesn’t support Redis cluster mode from memory(?) only hot failover. Harry will work on removing Redis from the GeoIP plugin. For group properties we could also use a local cache.

Last Sprint

- 🟡 making sure we don't drop person properties when there are updates across different distinct IDs @tiina303
  - Code clean-up done, still not done fully.
- 🟡 Librdkafka migration for analytics @xvello
  - Contribute nodes worker support to librdkafka
    - hopefully finished by eow
- 🟡 batch exports to S3 enabled for 3 teams @tomasfarias
  - Made progress, but didn't ship fully, blocked on PG migration

## Plan

- PoE: Rewrite squash to be based of Postgres data @tiina303
- Sentry PII @xvello
- Snowflake export enabled for a single customer @hazzadous

## OKRs

- Objective: CDP - @fuziontech
  - 🟢Key Result: Rock solid webhooks - @hazzadous 
  - 🟢Key Result: Rock solid batch processing (exports) - @tomasfarias
- Objective: Pipeline unblocks query performance and ensures data quality - @tiina303
  - 🟡Key Result: Persons on Events write path and squash shipped with monitoring and alerting
  - 🟡Key Result: Backfills are complete for all teams
  - 🔴Key Result: Guaranteed deduplication of UUIDs in events table within 7 days (with docs explaining it)
  - 🔴Key Result: Query time deduplication of events to cover until ClickHouse deduplication is complete
- Objective: Pipeline Robustness and Reliability - @xvello
  - 🟢Key Result: Errors cause kafka lag instead of DLQ production e.g. person creation failures cause kafka lag not go to DLQ
  - 🟢Key Result: Plugin server scales elastically and deploys do not create more than 2 minutes of lag

**Comment by neilkakkar** [comment]
# Team Feature Success

**Support hero:** @neilkakkar

## Retro

1. Flag autocapture + beta management follow up - @neilkakkar -> beta management is out for everyone; flag autocapture deprioritised for decide issues (rate limiting + db migration stuff) & some experimentation enhancements for $bigCustomer.
2. Productising feedback @liyiy -> MVP internal release blocked on db migration, but good progress, releasing for internal use on Monday.

## Hang over items from previous sprint

Flag autocapture - connecting to things in app


### High priority

1. Release surveys to beta users @liyiy 
Why? Core component of feature success. Allows getting qualitative user feedback quickly.
2. Dashboard templates on feature flags @EDsCODE 
Why? Allows diving deep into feature analytics
3. Decide instrumentation for pricing @neilkakkar 
Why? So we can start charging for flags

### Low priority / side quests

- Flag autocapture growth

**Comment by ellie** [comment]
# Team Infrastructure 

**Support hero:** @danielxnj 

## Retro

- @danielxnj Good support from the team with postgres, but frustrated we are still on it
- @ellie Great working with @xvello the past couple of days, but so many things breaking last week (loki, promtail, dev kafka etc)

## Hang over items from previous sprint



- Item 1. prioritised/deprioritise

## Plan
- @ellie: Deploy new capture to dev
- @ellie: Finish testing quickwit to replace loki, and otel collector setup (traces + logs)
- @ellie: Begin splitting up the chart so we don't redeploy all the things, all the time
- @danielxnj: Migrate postgres prod US
- @danielxnj + @ellie: Cost analysis and reduction

## OKRs

1. AWS Cost 🔴 increase in about 50%
2. AWS US prod migration from root account to dedicated account 🟡

**Comment by fuziontech** [comment]
# 💾 Team Data Warehouse 💾

**Support hero:** James

## Retro

- @fuziontech - Postgres and fires taking up most of my time this sprint.
- @frankh - Postgres was also very distracting - not much progress otherwise.

### Progress

- ✅ Get vector setup for Dev Ingestion @frankh 
  - Start with low risk ones
    - Partition statistics
    - Performance events
    - Meta columns in vector

## High priority
- Get vector setup for Prod Ingestion @frankh
  - Currently just reading events, want to consume all tables bound for CH
- Get one team reading from CH cloud (requires ^)  @frankh
- Clickhouse Copier - RACE (whoever has bandwidth first)
  - Use to get a few billion events in the S3 cluster
  - For query benchmarking purposes
- Upgrade CH Cluster version to 23.4 for cluster @fuziontech 
- Fiddle around with DBT - Open goal
- Backups @fuziontech 
  - Testing CH7 and CH8
  - Look into [disk backups](https://clickhouse.com/docs/en/operations/backup)
